cores/esp8266/Arduino.h

```diff
@@ -127,14 +127,8 @@ void timer0_isr_init(void);
 void timer0_attachInterrupt(timercallback userFunc);
 void timer0_detachInterrupt(void);
 
-// undefine stdlib's abs if encountered
-#ifdef abs
-#undef abs
-#endif
-
-#define abs(x) ((x)>0?(x):-(x))
+// Use stdlib abs() and round() to avoid issues with the C++ libraries
```

Comment:
*Sidenote:* since this is milestoned 3.0.0, can we push this change separately for 2.7.2?
Just noticed after switching back to master and also esp32 discrepancy https://github.com/espressif/arduino-esp32/commit/2f249edb8e2627421be87c43167a187e55ed7db7

btw I think `using abs;` is missing too?

Reply:
@d-a-v can comment more, as I think he tried this before and got nothing but grief from users of 3rd party libs about the change, but I think we had compatibility issues.  `std::abs` is very fussy about the operand types exactly matching, whereas the macro "just works".

If you'd really like it for 2.7.2, please open a separate issue to get his feedback.  My gut says it's too deep a change for a 3-dot release, but we'll see.

As for the missing `using std::abs;`, let me double check and update if needed.  Thx!

Reply:
Thx, fixed.

Reply:
I think I goofed up researching the issue. As full disclosure, [I was trying out building the test/host framework out of the tree, so I end up including mock Arduino.h....](https://github.com/mcspr/rpnlib/blob/master/examples/host/CMakeLists.txt) Which has quite different contents :)
Something to also note when running test suites!

Reply:
OK, if there's a host mock problem do open a bug on that so we can fix it, too!